solution/ui/prototype/src/views/Resources.vue

```diff
@@ -257,11 +262,44 @@ export default {
                 newQueryParams.title = "42"; // hard coding for now
                 newQueryParams[payload.scope] = payload.selectedIdentifier;
             }
+            if (payload.scope === "subpart") {
+                newQueryParams = await this.combineSections(
+                    payload.selectedIdentifier,
+                    newQueryParams
+                );
+                this.getPartDict(newQueryParams);
+            }
             this.$router.push({
                 name: "resources",
                 query: newQueryParams,
             });
         },
+
+        async combineSections(subpart, queryParams) {
+            let sections = await this.getSectionsBySubpart(subpart);
+            if (queryParams["section"]) {
+                sections = _uniq(
+                    queryParams["section"].split(",").concat(sections)
+                );
+            }
+
+            queryParams["section"] = sections.join(",");
+            return queryParams;
+        },
+        async getSectionsBySubpart(subpart) {
+            const splitSubpart = subpart.split("-");
+            const allSections = await getAllSections();
+            const sectionList = allSections
+                .filter((sec) => {
+                    return (
+                        sec.part === splitSubpart[0] &&
+                        sec.subpart == splitSubpart[1]
```

Comment:
mixing equality comparison (`==`) with strict equality comparison (`===`) -- is this intentional?